File: java/53 Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int n = nums.length;
        int max = Integer.MIN_VALUE;
        
        for(int i = 0; i < n; i++) {
            sum += nums[i];
            if (sum > max) max = sum;
            if (sum < 0) sum = 0;
        }
        return max;
    }
}

File: python/53 Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        res = float('-inf')
        curr = 0
        for i in range(len(nums)):
            curr = max(nums[i], nums[i] + curr)
            res = max(curr, res)
        return res
        
        """
        :type nums: List[int]
        :rtype: int
        """
        
